### DISCUSSION TITLE
使用FreeSql如何创建分表？

### DISCUSSION THREAD

**Discussion by issingle** [dotnetcore/FreeSql#801] [Q&A]
需要按月分表，如何使用codefirst或dbfirst创建 table_202101,table_202102....?

**Comment by luoyunchong** [comment]
ISelect  中 使用  AsTable(oldname => $"{oldname}_201903"); //对 Log_201903 表 CRUD

201903完全是自己定义的规则 。

code first 开启自动迁移，

FreeSqlBuilder 中**UseAutoSyncStructure(true)** 设置成true，开启自动迁移。

**Reply by issingle** [reply]
使用UseAutoSyncStructure是可以，但我不想启用自动同步，请问有通过代码手动的创建方法吗？

**Reply by luoyunchong** [reply]
https://github.com/dotnetcore/FreeSql/blob/master/FreeSql/Interface/ICodeFirst.cs#L84
用 `fsql.CodeFirst.SyncStructure(typeof(Song), "freesql_song")`
